File: app/src/main/java/org/thoughtcrime/securesms/service/webrtc/state/CallSetupState.java
package org.thoughtcrime.securesms.service.webrtc.state;

import androidx.annotation.NonNull;

/**
 * Information specific to setting up a call.
 */
public final class CallSetupState {
  boolean enableVideoOnCreate;
  boolean isRemoteVideoOffer;
  boolean acceptWithVideo;
  boolean sentJoinedMessage;
  boolean mobileDataAllowed;

  public CallSetupState() {
    this(false, false, false, false, false);
  }

  public CallSetupState(@NonNull CallSetupState toCopy) {
    this(toCopy.enableVideoOnCreate, toCopy.isRemoteVideoOffer, toCopy.acceptWithVideo, toCopy.sentJoinedMessage, toCopy.mobileDataAllowed);
  }

  public CallSetupState(boolean enableVideoOnCreate, boolean isRemoteVideoOffer, boolean acceptWithVideo, boolean sentJoinedMessage, boolean mobileDataAllowed) {
    this.enableVideoOnCreate = enableVideoOnCreate;
    this.isRemoteVideoOffer  = isRemoteVideoOffer;
    this.acceptWithVideo     = acceptWithVideo;
    this.sentJoinedMessage   = sentJoinedMessage;
    this.mobileDataAllowed   = mobileDataAllowed;
  }

  public boolean isEnableVideoOnCreate() {
    return enableVideoOnCreate;
  }

  public boolean isRemoteVideoOffer() {
    return isRemoteVideoOffer;
  }

  public boolean isAcceptWithVideo() {
    return acceptWithVideo;
  }

  public boolean hasSentJoinedMessage() {
    return sentJoinedMessage;
  }

  public boolean isMobileDataAllowed() {
    return mobileDataAllowed;
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/service/webrtc/state/CallSetupState.kt
package org.thoughtcrime.securesms.service.webrtc.state

import org.thoughtcrime.securesms.recipients.Recipient
import org.webrtc.PeerConnection

/**
 * Information specific to setting up a call.
 */
data class CallSetupState(
  var isEnableVideoOnCreate: Boolean = false,
  var isRemoteVideoOffer: Boolean = false,
  var isAcceptWithVideo: Boolean = false,
  @get:JvmName("hasSentJoinedMessage") var sentJoinedMessage: Boolean = false,
  @get:JvmName("shouldRingGroup") var ringGroup: Boolean = true,
  var ringId: Long = NO_RING,
  var ringerRecipient: Recipient = Recipient.UNKNOWN,
  @get:JvmName("shouldWaitForTelecomApproval") var waitForTelecom: Boolean = false,
  @get:JvmName("isTelecomApproved") var telecomApproved: Boolean = false,
  var iceServers: MutableList<PeerConnection.IceServer> = mutableListOf(),
  @get:JvmName("isAlwaysTurnServers") var alwaysTurnServers: Boolean = false
) {

  fun duplicate(): CallSetupState {
    return copy()
  }

  companion object {
    const val NO_RING = 0L
  }
}
